### PR TITLE
Fire o-tracking events when a marketing is shown

### PR DESCRIPTION
**Pull request by commuterjoy**
/cc @ifyio something like this?

### CODE REVIEW

File: src/prompter.js [hunk 1]
```diff
@@ -37,9 +37,24 @@ export default class Prompter{
         Prompter.setupDecorator(mConfig.Decorator, messenger);
         Prompter.uponClosingShutdown(messenger);
         messenger.showAfter(1500);
+		Prompter.track(prompt); // FIXME think about whether the 1500ms delay is a problem 
     }
 
-    static setupDecorator(Decorator, messenger) {
+	static track(prompt) {
+   		const data = {
+			action: 'marketing-message',
```
Comment (commuterjoy) [inline src/prompter.js]: might change this to 'message'

File: src/prompts/prompt/prompt.js [hunk 2]
```diff
@@ -25,6 +25,10 @@ export default class Prompt {
     get trackableAs() {
         return `${this.type}-prompt`;
     }
+    
+	get type() {
```
Comment (ifyio) [inline src/prompts/prompt/prompt.js]: Actually you can already do `prompt.type` on the instance so no need to add the getter here. Instead it seems like what needs to be done is to add a `type` property to https://github.com/Financial-Times/n-message-prompts/blob/master/src/prompts/propensity-messaging-prompt.js and ensure that it is populated with a value at https://github.com/Financial-Times/n-message-prompts/blob/master/src/prompts/propensity-messaging-prompt.js#L19

File: src/prompter.js [hunk 3]
```diff
@@ -37,9 +37,24 @@ export default class Prompter{
...
+			category: 'view',
+			context: {
+				product: 'next',
+				message: {
+					type: prompt.type()
```
Comment (ifyio) [inline src/prompter.js]: should be `prompt.type`

**Comment by commuterjoy** [comment]
@ifyio i made some changes. feel like i'm working blind though :)

File: src/lib/dispatch-custom-event.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export default function (name, data) {
+	const rootEl = document.body;
+	const event = (() => {
+		try {
+			return new CustomEvent(name, {bubbles: true, cancelable: true, detail: data});
```
Comment (matthew-andrews) [inline src/lib/dispatch-custom-event.js]: does the Polyfill service not sort this out for ya'll?

**Comment by commuterjoy** [comment]
o-tracking

ok, all working now.

**Pull request merged**